nwb_conversion_tools/datainterfaces/ecephys/neuralynx/neuralynxdatainterface.py

```diff
@@ -1,11 +1,72 @@
-"""Authors: Cody Baker."""
+"""Authors: Cody Baker and Ben Dichter."""
+import warnings
 from pathlib import Path
 from natsort import natsorted
+from glob import glob
+from dateutil import parser
+
 
 from spikeextractors import MultiRecordingChannelExtractor, NeuralynxRecordingExtractor
 
 from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
 from ....utils import FolderPathType
+from ....utils.json_schema import dict_deep_update
+
+
+def get_metadata(folder_path):
+    """
+    Parse the header of one of the .ncs files to get the session start time (without
+    timezone) and the session_id.
+
+    Parameters
+    ----------
+    folder_path: str
+
+    Returns
+    -------
+    dict
+
+    """
+    fpath = glob(folder_path + "/*.ncs")[0]
+    with open(fpath, "r", encoding="latin1") as file:
+        header = file.read(1024)
+    index = header.find("TimeCreated") + 12
+    session_start_time = parser.parse(header[index : index + 19])
+
+    index = header.find("SessionUUID") + 12
+    session_id = header[index : index + 36]
+
+    return dict(
+        session_start_time=session_start_time,
+        session_id=session_id,
+        identifier=session_id,
```

Comment:
```suggestion
```

Better to let the `identifier` get auto-set by us through `uuid` in the main converter, no?

Reply:
Or at the very least, generate another `uuid` at this line.

Reply:
why would we do that? a uuid will be generated if this is missing

Reply:
the session_id is guaranteed to be unique for each session. This could I guess lead to problems if you wanted to create two different NWB files for the same session

Reply:
In that case the user would manually override `metadata["NWBFile"]["session_id"]` if they wanted to, I simply don't want it to be possible for two NWBFiles made through this pipeline to have the same `identifier`, but if you're 100% sure that can't happen with the native Neuralynx ID here, then that's perfectly fine.